### PR TITLE
add backward-cpp

### PR DESCRIPTION
**Pull request by jj683**

### CODE REVIEW

File: packages/b/backward-cpp/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        local configs = {"-DBACKWARD_TESTS=OFF"}
+        table.insert(configs, "-DBACKWARD_SHARED=" .. (package:config("shared") and "ON" or "OFF"))
+        import("package.tools.cmake").build(package, configs, {buildir = "build_xmake"})
+
+        os.cp("backward.hpp", package:installdir("include/backward"))
+        if package:is_plat("windows") then
+            os.trycp(path.join("build_xmake", "*", "*.lib"), package:installdir("lib"))
+        else
+            os.trycp(path.join("build_xmake", "*.a"), package:installdir("lib"))
+        end
```
Comment (jj683) [inline packages/b/backward-cpp/xmake.lua:21]: By default, cmake copies "backward.hpp" without additional "backward" folder.

Comment (waruqi) [reply packages/b/backward-cpp/xmake.lua]: why not use cmake.install? no install scripts?

Comment (asafrnv) [reply packages/b/backward-cpp/xmake.lua]: We can, but "backward.hpp" will be located in the "root" folder and we have to include it like this:
`#include "backward.hpp"`

If it's not a problem, I'll use cmake.install.

Comment (jj683) [reply packages/b/backward-cpp/xmake.lua]: We can, but "backward.hpp" will be located in the "root" folder and we have to include it like this:
`#include "backward.hpp"`

If it's not a problem, I'll use cmake.install.

Comment (waruqi) [reply packages/b/backward-cpp/xmake.lua]: you can add `add_includedirs(".")` to rewrite default include dir

Comment (waruqi) [reply packages/b/backward-cpp/xmake.lua]: you can continue to copy/move header files after doing cmake.install

Comment (jj683) [reply packages/b/backward-cpp/xmake.lua]: Sounds good, will fix it soon.

**Pull request merged**